### ISSUE TITLE
no-shadowed-variablee should not warn on static methods

### ISSUE THREAD

**Issue by NN---** [palantir/tslint#3019]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.1
- __Running TSLint via__: gulp

#### TypeScript code being linted

```ts
class A<T> {
 static f<T>(x: T): void {} // error
}
```

with `tslint.json` configuration:

```json
{
  "rules": {"no-shadowed-variable": true }
}
```

#### Actual behavior
Complains on shadowed name 'T'

#### Expected behavior
No error since there is no shadowing.

**Comment by ajafff** [comment]
You're right, technically there's no shadowing.
One could argue that it would be easier to understand if you don't use the same name for both type parameters.

I don't have a strong opinion on that. It would be some effort to implement it, but it's doable.

**Comment by benny-medflyt** [comment]
I'm being hit by this as well.
I think it's actually better for readability for both type parameters to have the same name if they are both referring to the "same thing"

Would be great if this was fixed :+1:
